core/api/src/main/java/com/blazebit/persistence/CTEBuilder.java

```diff
@@ -36,6 +36,16 @@
      */
     public FullSelectCTECriteriaBuilder<T> with(Class<?> cteClass);
 
+    /**
+     * Creates a builder for a CTE with the given CTE type with the given criteria builder as basis.
+     *
+     * @param cteClass The type of the CTE
+     * @param criteriaBuilder The criteria builder to copy the query from
+     * @return The CTE builder
+     * @since 1.4.1
+     */
+    public FullSelectCTECriteriaBuilder<T> with(Class<?> cteClass, CriteriaBuilder<?> criteriaBuilder);
```

Comment:
At first I was confused as to what the difference is with `withCtesFrom`, but I think the idea behind this method is the following:

```suggestion
    public <X> FullSelectCTECriteriaBuilder<T> with(Class<X> cteClass, CriteriaBuilder<X> criteriaBuilder);
```

Right?

Reply:
No, the CTE type does not have to match the passed `CriteriaBuilder` as the attribute bindings have to be done explicitly.